src/traces/bar/set_positions.js

```diff
@@ -142,13 +143,15 @@ module.exports = function setPositions(gd, plotinfo) {
                 ti = gd.calcdata[bl[i]];
                 for(j = 0; j < ti.length; j++) {
                     sv = Math.round(ti[j].p / sumround);
+                    // store the negative sum value for p at the same key, with sign flipped
+                    if (relative && ti[j].s < 0) sv = -Math.round(ti[j].p / sumround);
```

Comment:
Maybe we could change this to `sv = -sv`. It's a super minor performance thing, but doesn't hurt  and also makes it a bit more immediately clear that it's _just_ the negative of the value.